http-client.ts

```diff
@@ -82,7 +82,16 @@ export class HttpClient implements Server.IHttpClient {
 				headers["Accept-Encoding"] = "gzip,deflate";
 			}
 
-			let result = new Future<Server.IResponse>();
+			let result = new Future<Server.IResponse>(),
+				timerId: number;
+
+			if (options.timeout) {
+				timerId = setTimeout(() => {
+					this.setResponseResult(result, timerId, { err: new Error(`Request to ${unmodifiedOptions.url} timed out.`) });
```

Comment:
I believe you can shorten this call a bit like this:
`this.setResponseResult(result, timerId, { response });`